contracts/TwabRewards.sol

```diff
@@ -137,19 +156,47 @@ contract TwabRewards is ITwabRewards {
     }
 
     /// @inheritdoc ITwabRewards
-    function cancelPromotion(uint256 _promotionId, address _to) external override returns (bool) {
+    function endPromotion(uint256 _promotionId, address _to) external override returns (bool) {
         require(_to != address(0), "TwabRewards/payee-not-zero-addr");
 
         Promotion memory _promotion = _getPromotion(_promotionId);
-        require(msg.sender == _promotion.creator, "TwabRewards/only-promo-creator");
+        _requirePromotionCreator(_promotion);
         _requirePromotionActive(_promotion);
 
-        _promotions[_promotionId].numberOfEpochs = uint8(_getCurrentEpochId(_promotion));
+        uint8 _epochNumber = uint8(_getCurrentEpochId(_promotion));
+        _promotions[_promotionId].numberOfEpochs = _epochNumber;
 
         uint256 _remainingRewards = _getRemainingRewards(_promotion);
         _promotion.token.safeTransfer(_to, _remainingRewards);
 
-        emit PromotionCancelled(_promotionId, _to, _remainingRewards);
+        emit PromotionEnded(_promotionId, _to, _remainingRewards, _epochNumber);
+
+        return true;
+    }
+
+    /// @inheritdoc ITwabRewards
+    function destroyPromotion(uint256 _promotionId, address _to) external override returns (bool) {
+        require(_to != address(0), "TwabRewards/payee-not-zero-addr");
+
+        Promotion memory _promotion = _getPromotion(_promotionId);
+        _requirePromotionCreator(_promotion);
+
+        require(
+            (_getPromotionEndTimestamp(_promotion) + _gracePeriod) < block.timestamp,
+            "TwabRewards/promotion-active"
```

Comment:
This is not accurate; the promotion may have ended but the grace period has not elapsed.

I think you should compare the **greater of** the promotion end or the created timestamp

i.e.

```
_requirePromotionInactive(_promotion)
graceEnd = (promotionEnd < createdAt ? createdAt : promotionEnd) + gracePeriod
require(block.timestamp >= graceEnd, "TwabRewards/grace-period-active")
```

That should give you the same information / errors

Reply:
Done: https://github.com/pooltogether/v4-periphery/pull/48/commits/f2fce3efe5158e62a50a68ff5c4e2fa1d9b0cf47